internal/rbd/replicationcontrollerserver.go

```diff
@@ -546,6 +551,35 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 	return &replication.PromoteVolumeResponse{}, nil
 }
 
+// checkHealthyPrimary checks if the is a healhty primary or not.
+// healthy primary image will be in up+stopped state, for states other
+// than this it returns an error message.
+func checkHealthyPrimary(ctx context.Context, rbdVol *rbdVolume) error {
+	mirrorStatus, err := rbdVol.getImageMirroringStatus()
+	if err != nil {
+		return err
```

Comment:
`return err` -> `return fmt.Errorf("failed to get mirror status: %w", err)` ?

Reply:
`getImageMirroringStatus` already returning the same error message. doing this will add a duplicate error message.